MetaMask/metamask-extension
# Refactor transaction ids to be UUIDs instead of random number

## Explanation


Core `TransactionController` creates UUID strings as an `id` for transactions. On the otherside extension creates random number while creating transactions using `createRandomId`. 

While we are on a mission for `TransactionController` alignment for both clients, for the sake of standardisation the use of UUIDs is preferred.

So this PR aims to make transaction ids to be UUID instead of random number. And also creates a migration for persisted transactions to migrate over UUIDs.

## Screenshots/Screencaps

No functional change.

## Manual Testing Steps



This PRs manual testing could be tricky since it's using a migration and needs a history of transactions persisted in previous commit. So testing could be done in these steps: 

1. Checkout `develop`
2. Send any transaction on any network (just to have some transaction history on the home page)
3. Checkout `1102-use-uuids-for-transaction-ids-in-extension-transaction-controller`
4. See history persist there without any issue (history should be clickable and detail modal should render without any issue)

Apart from the case above, we need to be able to navigate between transactions on multiple transaction notification:
<img width="250" alt="Screenshot 2023-09-18 at 12 52 24" src="https://github.com/MetaMask/metamask-extension/assets/7644512/0d3e8116-5c53-4c89-ac2f-ea3038a6fa6c">
 1. Try to send some transaction, don't close notification
 2. Open test-dapp on another tab, and do "Personal Sign" (basically creates another transaction)
 3. Try to navigate with the arrows on top of navigation without any issue
<img width="353" alt="image" src="https://github.com/MetaMask/metamask-extension/assets/7644512/10c168e8-1ced-4f67-904d-5205070a03ea">


## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
